### ISSUE TITLE
lab6

### ISSUE THREAD

**Issue by dlcor** [dlcor/AI#6]
import os
import cv2
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score, classification_report

# Параметры
data_dir = "cats_and_dogs_filtered"
img_size = (64, 64)  # Уменьшим размер, чтобы ускорить обучение

def load_images_from_folder(folder):
    images = []
    labels = []
    for label_name in ['cats', 'dogs']:
        label_folder = os.path.join(folder, label_name)
        label = 0 if label_name == 'cats' else 1
        for filename in os.listdir(label_folder):
            img_path = os.path.join(label_folder, filename)
            img = cv2.imread(img_path)
            if img is not None:
                img = cv2.resize(img, img_size)
                img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
                images.append(img)
                labels.append(label)
    return np.array(images), np.array(labels)

# Загрузка изображений
X_train, y_train = load_images_from_folder(os.path.join(data_dir, "train"))
X_val, y_val = load_images_from_folder(os.path.join(data_dir, "validation"))

# Визуализация примеров
plt.figure(figsize=(10, 10))
for i in range(9):
    plt.subplot(3, 3, i + 1)
    plt.imshow(X_train[i])
    plt.title("Cat" if y_train[i] == 0 else "Dog")
    plt.axis("off")
plt.show()

# Предобработка данных
X_train_flat = X_train.reshape((X_train.shape[0], -1)) / 255.0
X_val_flat = X_val.reshape((X_val.shape[0], -1)) / 255.0

# Обучение модели
model = SVC(kernel='linear')  # Или попробуй 'rbf'
model.fit(X_train_flat, y_train)

# Оценка
y_pred = model.predict(X_val_flat)
accuracy = accuracy_score(y_val, y_pred)
print(f"Accuracy: {accuracy:.4f}")
print(classification_report(y_val, y_pred, target_names=["Cat", "Dog"]))

# Визуализация некоторых предсказаний
plt.figure(figsize=(10, 10))
for i in range(9):
    plt.subplot(3, 3, i + 1)
    plt.imshow(X_val[i])
    plt.title(f"Pred: {'Cat' if y_pred[i]==0 else 'Dog'}")
    plt.axis("off")
plt.show()
